central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -646,3 +646,17 @@ func (ds *datastoreImpl) RemovePLOPsWithoutProcessIndicatorOrProcessInfo(ctx con
 
 	return int64(len(plopsToDelete)), nil
 }
+
+func (ds *datastoreImpl) RemovePLOPsWithoutPodUID(ctx context.Context) (int64, error) {
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	deletedPlops, err := ds.storage.DeleteByQuery(ctx, search.NewQueryBuilder().
```

Comment:
We don't need all ids but I see we don't have a better alternative. Since it's not called often I think it's fine.

Reply:
This goes back to my original concern.  The risk of doing this in all at once in a delete statement is that if for some reason the delete fails (timeout, deadlock, etc) then nothing will get deleted.  If we query to get the IDs and then run batching of deleting the IDs AND don't rollback on an error we will eventually get rid of the data.  We've seen this cause problems before trying to prune process indicators by just running it as a delete.

I'm not a fan of 2-pass, but what we've seen based on our current implementation and usage is that getting the list of things to delete and then deleting them has been more successful

Reply:
We also have this aproach https://github.com/stackrox/stackrox/blob/f6a2391b96cbf8be384b9d5592e729463c7cef03/central/postgres/pruning.go
Why not using it here?

Reply:
I am now using something more like the approach in the link. I am getting the ids in a Retry2 and then deleting them in batches.

Reply:
I now have something that deletes in batches without fetching any data to central, except the number of deleted rows.